app/javascript/serviceworker/messages.spec.js

```diff
@@ -31,16 +31,32 @@ describe("messageHandler", () => {
   });
 
   describe("SAVE_CAMPAIGN_FOR_OFFLINE", () => {
-    test("works", () => {
-      handler(event("SAVE_CAMPAIGN_FOR_OFFLINE", ["1"]));
+    test("works", async () => {
+      handler(
+        event("SAVE_CAMPAIGN_FOR_OFFLINE", {
+          password: "test",
+          campaignId: 1,
+          additionalItems: ["/assets/application.js"],
+        })
+      );
+
+      await Promise.resolve(); // Wait for initCache to finish
+      await Promise.resolve(); // Wait for addAll to finish
+
+      expect(initCache).toHaveBeenCalledWith("test");
       expect(addAll.mock.calls[0][0]).toMatchInlineSnapshot(`
         [
-          "/campaigns/undefined/children",
-          "/campaigns/undefined/children.json",
-          "/campaigns/undefined/children/record-template",
-          "/campaigns/undefined/children/show-template",
+          "/assets/application.js",
+          "/favicon.ico",
+          "/dashboard",
```

Comment:
This part concerns me a little ... I think the access codes are per-campaign. It's possible a SAIS nurse will be working on more than one offline campaign at a time, esp if they're part of a big team. I think for security purposes, seeing as each campaign is a different data set we should have a different access code for each campaign. 

That being the case, I don't think assets that are used across assets should be encrypted. For example, which password would they enter to view `/dashboard`?

In any case, probably for another PR, I'll create an issue.